### PR TITLE
Fix shamir RFC

### PR DESCRIPTION
**Pull request by TimeEngineer**
# What has changed ?

`self` is a rust keyword which isn't allowed as raw identifier, see: https://github.com/rust-lang/rust/blob/1.31.1/src/libsyntax_pos/symbol.rs#L455-L463

- [ ] Does this PR affect the User ?

### CODE REVIEW

File: docs/rfcs/1000-shamir-based-recovery.md [hunk 1]
```diff
@@ -267,21 +267,21 @@ Authenticated API:
...
         "reps": [
             {
                 "status": "ok",
-                "other_fields": [
+                "fields": [
                     {
                         // `None` means no configuration
                         // Otherwise, contains a `ShamirRecoveryBriefCertificate`
-                        "name": "self",
+                        "name": "my",
```
Comment (FirelightFlagboy) [inline docs/rfcs/1000-shamir-based-recovery.md:279]: `info` or `my_info` ?

Comment (TimeEngineer) [reply docs/rfcs/1000-shamir-based-recovery.md:279]: I can also set to `self_info`

File: docs/rfcs/1000-shamir-based-recovery.md [hunk 2]
```diff
@@ -430,7 +430,7 @@ Invited API, we reuse the `invite_info` command:
                             },
                             {
                                 "name": "recipients",
-                                "type": "Vec<ShamirRecoveryRecipient>",
+                                "type": "Vec<ShamirRecoveryRecipient>"
```
Comment (touilleMan) [inline docs/rfcs/1000-shamir-based-recovery.md:433]: Not a `List<...>` here ?

**Review by touilleMan — APPROVED** [review]

File: docs/rfcs/1000-shamir-based-recovery.md (hunk 2 above)
Comment (TimeEngineer) [reply docs/rfcs/1000-shamir-based-recovery.md:433]: Sure

**Pull request merged**